# Titanic Analysis

## Introduction

The Titanic dataset contains information on 891 passengers aboard the RMS Titanic, including demographics, travel details, and survival status. It is commonly used to explore which factors—such as age, gender, and passenger class—affected the likelihood of survival. This dataset provides a simple yet powerful foundation for learning data analysis and building predictive models.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
import warnings
warnings.filterwarnings('ignore')

In [3]:
df=pd.read_csv('Titanic-Dataset.csv')
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

## Data Description
* PassengerID: Unique identifier for each passenger.
* Pclass: Passenger class (1st, 2nd, 3rd).
* Name: Full name of the passenger.
* Sex: Gender of the passenger.
* Age: Age of the passenger.
* SibSp: Number of siblings/spouses aboard.
* nParch: Number of parents/children aboard.
* Ticket: Ticket number.
* Fare: Ticket fare.
* Cabin: Cabin number.
* Embarked: Port of embarkation (C = Cherbourg; Q = Queenstown; S = Southampton).
* Survived: Survival status (0 = No; 1 = Yes).

In [4]:
df.size

10692

In [5]:
df.shape

(891, 12)

In [6]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [8]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

### Information of the Dataset
* Columns = 12
* Rows = 891
* Data types
    * integer = 5
    * float = 2
    * object = 5
* Occur Null values
    * Age = 177
    * Cabin = 687
    * Embarked = 2

In [10]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

### Fill Null Values

In [11]:
df['Age'] = df['Age'].fillna(df['Age'].median())

In [12]:
df['Cabin'] = df['Cabin'].fillna('Unknown')

In [13]:
df['Embarked'] = df['Embarked'].fillna(df['Embarked'].mode()[0])

In [14]:
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Cabin          0
Embarked       0
dtype: int64

## Pandas Filtering Questions

#### 1) Get the details of passengers who survived and were younger than 18 years old.

In [16]:
df[(df['Survived']==1) & (df['Age']<18)]

PassengerId  Survived  Pclass                                      Name  \
9             10         1       2       Nasser, Mrs. Nicholas (Adele Achem)   
10            11         1       3           Sandstrom, Miss. Marguerite Rut   
22            23         1       3               McGowan, Miss. Anna "Annie"   
39            40         1       3               Nicola-Yarred, Miss. Jamila   
43            44         1       2  Laroche, Miss. Simonne Marie Anne Andree   
..           ...       ...     ...                                       ...   
830          831         1       3   Yasbeck, Mrs. Antoni (Selini Alexander)   
831          832         1       2           Richards, Master. George Sibley   
853          854         1       1                 Lines, Miss. Mary Conover   
869          870         1       3           Johnson, Master. Harold Theodor   
875          876         1       3          Najib, Miss. Adele Kiamie "Jane"   

        Sex    Age  SibSp  Parch         Ticket     Fare    Cabin Embarked  
9    female  14.00      1      0         237736  30.0708  Unknown        C  
10   female   4.00      1      1        PP 9549  16.7000       G6        S  
22   female  15.00      0      0         330923   8.0292  Unknown        Q  
39   female  14.00      1      0           2651  11.2417  Unknown        C  
43   female   3.00      1      2  SC/Paris 2123  41.5792  Unknown        C  
..      ...    ...    ...    ...            ...      ...      ...      ...  
830  female  15.00      1      0           2659  14.4542  Unknown        C  
831    male   0.83      1      1          29106  18.7500  Unknown        S  
853  female  16.00      0      1       PC 17592  39.4000      D28        S  
869    male   4.00      1      1         347742  11.1333  Unknown        S  
875  female  15.00      0      0           2667   7.2250  Unknown        C  

[61 rows x 12 columns]

#### 2) Get the details of male passengers who were older than 40 years old

In [17]:
df[(df['Sex']=='male') & (df['Age']>40)]

PassengerId  Survived  Pclass                            Name   Sex  \
6              7         0       1         McCarthy, Mr. Timothy J  male   
33            34         0       2           Wheadon, Mr. Edward H  male   
35            36         0       1  Holverson, Mr. Alexander Oskar  male   
54            55         0       1  Ostby, Mr. Engelhart Cornelius  male   
62            63         0       1     Harris, Mr. Henry Birkhardt  male   
..           ...       ...     ...                             ...   ...   
845          846         0       3             Abbing, Mr. Anthony  male   
851          852         0       3             Svensson, Mr. Johan  male   
857          858         1       1          Daly, Mr. Peter Denis   male   
860          861         0       3         Hansen, Mr. Claus Peter  male   
873          874         0       3     Vander Cruyssen, Mr. Victor  male   

      Age  SibSp  Parch      Ticket     Fare    Cabin Embarked  
6    54.0      0      0       17463  51.8625      E46        S  
33   66.0      0      0  C.A. 24579  10.5000  Unknown        S  
35   42.0      1      0      113789  52.0000  Unknown        S  
54   65.0      0      1      113509  61.9792      B30        C  
62   45.0      1      0       36973  83.4750      C83        S  
..    ...    ...    ...         ...      ...      ...      ...  
845  42.0      0      0   C.A. 5547   7.5500  Unknown        S  
851  74.0      0      0      347060   7.7750  Unknown        S  
857  51.0      0      0      113055  26.5500      E17        S  
860  41.0      2      0      350026  14.1083  Unknown        S  
873  47.0      0      0      345765   9.0000  Unknown        S  

[102 rows x 12 columns]

#### 3) Get the details of passengers who paid a fare greater than $30 and survived.

In [18]:
df[(df['Fare']>30) & (df['Survived']==1)]

PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
9             10         1       2   
23            24         1       1   
25            26         1       3   
..           ...       ...     ...   
849          850         1       1   
853          854         1       1   
856          857         1       1   
871          872         1       1   
879          880         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
23                        Sloper, Mr. William Thompson    male  28.0      0   
25   Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female  38.0      1   
..                                                 ...     ...   ...    ...   
849       Goldenberg, Mrs. Samuel L (Edwiga Grabowska)  female  28.0      1   
853                          Lines, Miss. Mary Conover  female  16.0      0   
856         Wick, Mrs. George Dennick (Mary Hitchcock)  female  45.0      1   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   

     Parch    Ticket      Fare    Cabin Embarked  
1        0  PC 17599   71.2833      C85        C  
3        0    113803   53.1000     C123        S  
9        0    237736   30.0708  Unknown        C  
23       0    113788   35.5000       A6        S  
25       5    347077   31.3875  Unknown        S  
..     ...       ...       ...      ...      ...  
849      0     17453   89.1042      C92        C  
853      1  PC 17592   39.4000      D28        S  
856      1     36928  164.8667  Unknown        S  
871      1     11751   52.5542      D35        S  
879      1     11767   83.1583      C50        C  

[136 rows x 12 columns]

#### 4) Get the details of female passengers who did not survive.

In [19]:
df[(df['Sex']=='female') & (df['Survived']==0)]

PassengerId  Survived  Pclass  \
14            15         0       3   
18            19         0       3   
24            25         0       3   
38            39         0       3   
40            41         0       3   
..           ...       ...     ...   
854          855         0       2   
863          864         0       3   
882          883         0       3   
885          886         0       3   
888          889         0       3   

                                                  Name     Sex   Age  SibSp  \
14                Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   
18   Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
24                       Palsson, Miss. Torborg Danira  female   8.0      3   
38                  Vander Planke, Miss. Augusta Maria  female  18.0      2   
40      Ahlin, Mrs. Johan (Johanna Persdotter Larsson)  female  40.0      1   
..                                                 ...     ...   ...    ...   
854      Carter, Mrs. Ernest Courtenay (Lilian Hughes)  female  44.0      1   
863                  Sage, Miss. Dorothy Edith "Dolly"  female  28.0      8   
882                       Dahlberg, Miss. Gerda Ulrika  female  22.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female  28.0      1   

     Parch      Ticket     Fare    Cabin Embarked  
14       0      350406   7.8542  Unknown        S  
18       0      345763  18.0000  Unknown        S  
24       1      349909  21.0750  Unknown        S  
38       0      345764  18.0000  Unknown        S  
40       0        7546   9.4750  Unknown        S  
..     ...         ...      ...      ...      ...  
854      0      244252  26.0000  Unknown        S  
863      2    CA. 2343  69.5500  Unknown        S  
882      0        7552  10.5167  Unknown        S  
885      5      382652  29.1250  Unknown        Q  
888      2  W./C. 6607  23.4500  Unknown        S  

[81 rows x 12 columns]

#### 5) Get the details of passengers who were in either the first or second class.

In [20]:
df[(df['Pclass']==1) | (df['Pclass']==2)]

PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
6              7         0       1   
9             10         1       2   
11            12         1       1   
..           ...       ...     ...   
880          881         1       2   
883          884         0       2   
886          887         0       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
..                                                 ...     ...   ...    ...   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
883                      Banfield, Mr. Frederick James    male  28.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare    Cabin Embarked  
1        0          PC 17599  71.2833      C85        C  
3        0            113803  53.1000     C123        S  
6        0             17463  51.8625      E46        S  
9        0            237736  30.0708  Unknown        C  
11       0            113783  26.5500     C103        S  
..     ...               ...      ...      ...      ...  
880      1            230433  26.0000  Unknown        S  
883      0  C.A./SOTON 34068  10.5000  Unknown        S  
886      0            211536  13.0000  Unknown        S  
887      0            112053  30.0000      B42        S  
889      0            111369  30.0000     C148        C  

[400 rows x 12 columns]

#### 6) Get the details of passengers who had more than two siblings/spouses aboard the        Titanic and did not survive.

In [21]:
df[(df['SibSp']>2) & (df['Survived']==0)]

PassengerId  Survived  Pclass                                     Name  \
7              8         0       3           Palsson, Master. Gosta Leonard   
16            17         0       3                     Rice, Master. Eugene   
24            25         0       3            Palsson, Miss. Torborg Danira   
27            28         0       1           Fortune, Mr. Charles Alexander   
50            51         0       3               Panula, Master. Juha Niilo   
59            60         0       3       Goodwin, Master. William Frederick   
63            64         0       3                    Skoog, Master. Harald   
71            72         0       3               Goodwin, Miss. Lillian Amy   
119          120         0       3        Andersson, Miss. Ellis Anna Maria   
159          160         0       3               Sage, Master. Thomas Henry   
164          165         0       3             Panula, Master. Eino Viljami   
171          172         0       3                     Rice, Master. Arthur   
176          177         0       3            Lefebre, Master. Henry Forbes   
180          181         0       3             Sage, Miss. Constance Gladys   
182          183         0       3    Asplund, Master. Clarence Gustaf Hugo   
201          202         0       3                      Sage, Mr. Frederick   
229          230         0       3                  Lefebre, Miss. Mathilde   
266          267         0       3                Panula, Mr. Ernesti Arvid   
278          279         0       3                       Rice, Master. Eric   
324          325         0       3                 Sage, Mr. George John Jr   
374          375         0       3               Palsson, Miss. Stina Viola   
386          387         0       3          Goodwin, Master. Sidney Leonard   
409          410         0       3                       Lefebre, Miss. Ida   
480          481         0       3           Goodwin, Master. Harold Victor   
485          486         0       3                   Lefebre, Miss. Jeannie   
541          542         0       3     Andersson, Miss. Ingeborg Constanzia   
542          543         0       3        Andersson, Miss. Sigrid Elisabeth   
634          635         0       3                       Skoog, Miss. Mabel   
642          643         0       3            Skoog, Miss. Margit Elizabeth   
683          684         0       3              Goodwin, Mr. Charles Edward   
686          687         0       3                 Panula, Mr. Jaako Arnold   
787          788         0       3                Rice, Master. George Hugh   
792          793         0       3                  Sage, Miss. Stella Anna   
813          814         0       3       Andersson, Miss. Ebba Iris Alfrida   
819          820         0       3             Skoog, Master. Karl Thorsten   
824          825         0       3             Panula, Master. Urho Abraham   
846          847         0       3                 Sage, Mr. Douglas Bullen   
850          851         0       3  Andersson, Master. Sigvard Harald Elias   
863          864         0       3        Sage, Miss. Dorothy Edith "Dolly"   

        Sex   Age  SibSp  Parch    Ticket      Fare        Cabin Embarked  
7      male   2.0      3      1    349909   21.0750      Unknown        S  
16     male   2.0      4      1    382652   29.1250      Unknown        Q  
24   female   8.0      3      1    349909   21.0750      Unknown        S  
27     male  19.0      3      2     19950  263.0000  C23 C25 C27        S  
50     male   7.0      4      1   3101295   39.6875      Unknown        S  
59     male  11.0      5      2   CA 2144   46.9000      Unknown        S  
63     male   4.0      3      2    347088   27.9000      Unknown        S  
71   female  16.0      5      2   CA 2144   46.9000      Unknown        S  
119  female   2.0      4      2    347082   31.2750      Unknown        S  
159    male  28.0      8      2  CA. 2343   69.5500      Unknown        S  
164    m

#### 7) Get the details of passengers who were in the third class and were older than 30 years old.

In [22]:
df[(df['Pclass']==3) & (df['Age']>30)]

PassengerId  Survived  Pclass  \
4              5         0       3   
13            14         0       3   
18            19         0       3   
25            26         1       3   
40            41         0       3   
..           ...       ...     ...   
860          861         0       3   
873          874         0       3   
881          882         0       3   
885          886         0       3   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
4                             Allen, Mr. William Henry    male  35.0      0   
13                         Andersson, Mr. Anders Johan    male  39.0      1   
18   Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
25   Asplund, Mrs. Carl Oscar (Selma Augusta Emilia...  female  38.0      1   
40      Ahlin, Mrs. Johan (Johanna Persdotter Larsson)  female  40.0      1   
..                                                 ...     ...   ...    ...   
860                            Hansen, Mr. Claus Peter    male  41.0      2   
873                        Vander Cruyssen, Mr. Victor    male  47.0      0   
881                                 Markun, Mr. Johann    male  33.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch  Ticket     Fare    Cabin Embarked  
4        0  373450   8.0500  Unknown        S  
13       5  347082  31.2750  Unknown        S  
18       0  345763  18.0000  Unknown        S  
25       5  347077  31.3875  Unknown        S  
40       0    7546   9.4750  Unknown        S  
..     ...     ...      ...      ...      ...  
860      0  350026  14.1083  Unknown        S  
873      0  345765   9.0000  Unknown        S  
881      0  349257   7.8958  Unknown        S  
885      5  382652  29.1250  Unknown        Q  
890      0  370376   7.7500  Unknown        Q  

[103 rows x 12 columns]

#### 8) Get the details of passengers who were younger than 15 years old and had parents/children aboard.

In [23]:
df[(df['Age']<15) & (df['Parch']>0)]

PassengerId  Survived  Pclass                                      Name  \
7              8         0       3            Palsson, Master. Gosta Leonard   
10            11         1       3           Sandstrom, Miss. Marguerite Rut   
16            17         0       3                      Rice, Master. Eugene   
24            25         0       3             Palsson, Miss. Torborg Danira   
43            44         1       2  Laroche, Miss. Simonne Marie Anne Andree   
..           ...       ...     ...                                       ...   
827          828         1       2                     Mallet, Master. Andre   
831          832         1       2           Richards, Master. George Sibley   
850          851         0       3   Andersson, Master. Sigvard Harald Elias   
852          853         0       3                   Boulos, Miss. Nourelain   
869          870         1       3           Johnson, Master. Harold Theodor   

        Sex   Age  SibSp  Parch           Ticket     Fare    Cabin Embarked  
7      male  2.00      3      1           349909  21.0750  Unknown        S  
10   female  4.00      1      1          PP 9549  16.7000       G6        S  
16     male  2.00      4      1           382652  29.1250  Unknown        Q  
24   female  8.00      3      1           349909  21.0750  Unknown        S  
43   female  3.00      1      2    SC/Paris 2123  41.5792  Unknown        C  
..      ...   ...    ...    ...              ...      ...      ...      ...  
827    male  1.00      0      2  S.C./PARIS 2079  37.0042  Unknown        C  
831    male  0.83      1      1            29106  18.7500  Unknown        S  
850    male  4.00      4      2           347082  31.2750  Unknown        S  
852  female  9.00      1      1             2678  15.2458  Unknown        C  
869    male  4.00      1      1           347742  11.1333  Unknown        S  

[70 rows x 12 columns]

#### 9) Get the details of male passengers who paid less than $10 for their fare.

In [24]:
df[(df['Sex']=='male') & (df['Fare']<10)]

PassengerId  Survived  Pclass                            Name   Sex  \
0              1         0       3         Braund, Mr. Owen Harris  male   
4              5         0       3        Allen, Mr. William Henry  male   
5              6         0       3                Moran, Mr. James  male   
12            13         0       3  Saundercock, Mr. William Henry  male   
26            27         0       3         Emir, Mr. Farred Chehab  male   
..           ...       ...     ...                             ...   ...   
877          878         0       3            Petroff, Mr. Nedelio  male   
878          879         0       3              Laleff, Mr. Kristo  male   
881          882         0       3              Markun, Mr. Johann  male   
884          885         0       3          Sutehall, Mr. Henry Jr  male   
890          891         0       3             Dooley, Mr. Patrick  male   

      Age  SibSp  Parch           Ticket    Fare    Cabin Embarked  
0    22.0      1      0        A/5 21171  7.2500  Unknown        S  
4    35.0      0      0           373450  8.0500  Unknown        S  
5    28.0      0      0           330877  8.4583  Unknown        Q  
12   20.0      0      0        A/5. 2151  8.0500  Unknown        S  
26   28.0      0      0             2631  7.2250  Unknown        C  
..    ...    ...    ...              ...     ...      ...      ...  
877  19.0      0      0           349212  7.8958  Unknown        S  
878  28.0      0      0           349217  7.8958  Unknown        S  
881  33.0      0      0           349257  7.8958  Unknown        S  
884  25.0      0      0  SOTON/OQ 392076  7.0500  Unknown        S  
890  32.0      0      0           370376  7.7500  Unknown        Q  

[272 rows x 12 columns]

#### 10) Get the details of passengers who were in the third class, younger than 20, and had siblings/spouses aboard.

In [25]:
df[(df['Pclass']==3) & (df['Age']<20) & (df['SibSp']>0)]

PassengerId  Survived  Pclass  \
7              8         0       3   
10            11         1       3   
16            17         0       3   
24            25         0       3   
38            39         0       3   
39            40         1       3   
49            50         0       3   
50            51         0       3   
59            60         0       3   
63            64         0       3   
68            69         1       3   
71            72         0       3   
86            87         0       3   
111          112         0       3   
119          120         0       3   
125          126         1       3   
147          148         0       3   
164          165         0       3   
171          172         0       3   
172          173         1       3   
175          176         0       3   
182          183         0       3   
192          193         1       3   
233          234         1       3   
261          262         1       3   
266          267         0       3   
278          279         0       3   
333          334         0       3   
348          349         1       3   
352          353         0       3   
371          372         0       3   
374          375         0       3   
386          387         0       3   
424          425         0       3   
448          449         1       3   
469          470         1       3   
480          481         0       3   
489          490         1       3   
532          533         0       3   
541          542         0       3   
542          543         0       3   
634          635         0       3   
642          643         0       3   
644          645         1       3   
683          684         0       3   
686          687         0       3   
721          722         0       3   
746          747         0       3   
787          788         0       3   
788          789         1       3   
813          814         0       3   
819          820         0       3   
824          825         0       3   
830          831         1       3   
850          851         0       3   
852          853         0       3   
869          870         1       3   

                                                Name     Sex    Age  SibSp  \
7                     Palsson, Master. Gosta Leonard    male   2.00      3   
10                   Sandstrom, Miss. Marguerite Rut  female   4.00      1   
16                              Rice, Master. Eugene    male   2.00      4   
24                     Palsson, Miss. Torborg Danira  female   8.00      3   
38                Vander Planke, Miss. Augusta Maria  female  18.00      2   
39                       Nicola-Yarred, Miss. Jamila  female  14.00      1   
49     Arnold-Franchi, Mrs. Josef (Josefine Franchi)  female  18.00      1   
50                        Panula, Master. Juha Niilo    male   7.00      4   
59                Goodwin, Master. William Frederick    male  11.00      5   
63                             Skoog, Master. Harald    male   4.00      3   
68                   Andersson, Miss. Erna Alexandra  female  17.00      4   
71                        Goodwin, Miss. Lillian Amy  female  16.00      5   
86                            Ford, Mr. William Neal    male  16.00      1   
111                             Zabour, Miss. Hileni  female  14.50      1   
119                Andersson, Miss. Ellis Anna Maria  female   2.00      4   
125                     Nicola-Yarred, Master. Elias    male  12.00      1   
147                 Ford, Miss. Robina Maggie "Ruby"  female   9.00      2   
164                     Panula, Master. Eino Viljami    male   1.00      4   
171                             Rice, Master. Arthur    male   4.00      4   
172                     Johnson, Miss. Eleanor Ileen  female   1.00      1   
175                           Klasen, Mr. Klas Albin    male  18.00      1   
182            Asplund, Master. Clarence Gustaf Hugo    male   9.00      4   
192  A

#### 11) Get the details of passengers who were either younger than 10 years or older than 60 years, and survived.

In [26]:
df[((df['Age']<10) | (df['Age']>60)) & (df['Survived']==1)]

PassengerId  Survived  Pclass  \
10            11         1       3   
43            44         1       2   
58            59         1       2   
78            79         1       2   
165          166         1       3   
172          173         1       3   
183          184         1       2   
184          185         1       3   
193          194         1       2   
233          234         1       3   
237          238         1       2   
261          262         1       3   
275          276         1       1   
305          306         1       1   
340          341         1       2   
348          349         1       3   
381          382         1       3   
407          408         1       2   
445          446         1       1   
448          449         1       3   
469          470         1       3   
479          480         1       3   
483          484         1       3   
489          490         1       3   
530          531         1       2   
535          536         1       2   
549          550         1       2   
570          571         1       2   
618          619         1       2   
630          631         1       1   
644          645         1       3   
691          692         1       3   
720          721         1       2   
750          751         1       2   
751          752         1       3   
755          756         1       2   
777          778         1       3   
788          789         1       3   
803          804         1       3   
827          828         1       2   
829          830         1       1   
831          832         1       2   
869          870         1       3   

                                                Name     Sex    Age  SibSp  \
10                   Sandstrom, Miss. Marguerite Rut  female   4.00      1   
43          Laroche, Miss. Simonne Marie Anne Andree  female   3.00      1   
58                      West, Miss. Constance Mirium  female   5.00      1   
78                     Caldwell, Master. Alden Gates    male   0.83      0   
165  Goldsmith, Master. Frank John William "Frankie"    male   9.00      0   
172                     Johnson, Miss. Eleanor Ileen  female   1.00      1   
183                        Becker, Master. Richard F    male   1.00      2   
184              Kink-Heilmann, Miss. Luise Gretchen  female   4.00      0   
193                       Navratil, Master. Michel M    male   3.00      1   
233                   Asplund, Miss. Lillian Gertrud  female   5.00      4   
237                 Collyer, Miss. Marjorie "Lottie"  female   8.00      0   
261                Asplund, Master. Edvin Rojj Felix    male   3.00      4   
275                Andrews, Miss. Kornelia Theodosia  female  63.00      1   
305                   Allison, Master. Hudson Trevor    male   0.92      1   
340                   Navratil, Master. Edmond Roger    male   2.00      1   
348           Coutts, Master. William Loch "William"    male   3.00      1   
381                      Nakid, Miss. Maria ("Mary")  female   1.00      0   
407                   Richards, Master. William Rowe    male   3.00      1   
445                        Dodge, Master. Washington    male   4.00      0   
448                   Baclini, Miss. Marie Catherine  female   5.00      2   
469                    Baclini, Miss. Helene Barbara  female   0.75      2   
479                         Hirvonen, Miss. Hildur E  female   2.00      0   
483                           Turkula, Mrs. (Hedwig)  female  63.00      0   
489            Coutts, Master. Eden Leslie "Neville"    male   9.00      1   
530                         Quick, Miss. Phyllis May  female   2.00      1   
535                           Hart, Miss. Eva Miriam  female   7.00      0   
549                   Davies, Master. John Morgan Jr    male   8.00      1   
570                               Harris, Mr. George    male  62.00      0   
618                      Becker, Miss. Marion Louise  female   4.00   

#### 12) get the name, passenger id and ticket no. who were female and embarked in Southampton

In [27]:
df[(df['Sex']=='female') & (df['Embarked']=='S')][['Name','PassengerId','Ticket']]

Name  PassengerId  \
2                               Heikkinen, Miss. Laina            3   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)            4   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)            9   
10                     Sandstrom, Miss. Marguerite Rut           11   
11                            Bonnell, Miss. Elizabeth           12   
..                                                 ...          ...   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)          872   
880       Shelley, Mrs. William (Imanita Parrish Hall)          881   
882                       Dahlberg, Miss. Gerda Ulrika          883   
887                       Graham, Miss. Margaret Edith          888   
888           Johnston, Miss. Catherine Helen "Carrie"          889   

               Ticket  
2    STON/O2. 3101282  
3              113803  
8              347742  
10            PP 9549  
11             113783  
..                ...  
871             11751  
880            230433  
882              7552  
887            112053  
888        W./C. 6607  

[205 rows x 3 columns]

#### 13) get the names of male pssengers who were survived.

In [28]:
df[(df['Sex']=='male') & (df['Survived']==1)]['Name']

17        Williams, Mr. Charles Eugene
21               Beesley, Mr. Lawrence
23        Sloper, Mr. William Thompson
36                    Mamee, Mr. Hanna
55                   Woolner, Mr. Hugh
                    ...               
838                    Chip, Mr. Chang
839               Marechal, Mr. Pierre
857             Daly, Mr. Peter Denis 
869    Johnson, Master. Harold Theodor
889              Behr, Mr. Karl Howell
Name: Name, Length: 109, dtype: object

#### 14) get the name, passenger id and cabin no. who were in the second class and the ticket fare greater than 50

In [29]:
df[(df['Pclass']==2) & (df['Fare']>50)][['Name','PassengerId','Cabin']]

Name  PassengerId    Cabin
72               Hood, Mr. Ambrose Jr           73  Unknown
120       Hickman, Mr. Stanley George          121  Unknown
385         Davies, Mr. Charles Henry          386  Unknown
615               Herman, Miss. Alice          616  Unknown
655         Hickman, Mr. Leonard Mark          656  Unknown
665                Hickman, Mr. Lewis          666  Unknown
754  Herman, Mrs. Samuel (Jane Laver)          755  Unknown

#### 15) get the details of passengers who were embarked in cherbourg or older than 50 years

In [30]:
df[(df['Embarked']=='C') | (df['Age']>50)]

PassengerId  Survived  Pclass  \
1              2         1       1   
6              7         0       1   
9             10         1       2   
11            12         1       1   
15            16         1       2   
..           ...       ...     ...   
866          867         1       2   
874          875         1       2   
875          876         1       3   
879          880         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
15                    Hewlett, Mrs. (Mary D Kingcome)   female  55.0      0   
..                                                 ...     ...   ...    ...   
866                       Duran y More, Miss. Asuncion  female  27.0      1   
874              Abelson, Mrs. Samuel (Hannah Wizosky)  female  28.0      1   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch         Ticket     Fare    Cabin Embarked  
1        0       PC 17599  71.2833      C85        C  
6        0          17463  51.8625      E46        S  
9        0         237736  30.0708  Unknown        C  
11       0         113783  26.5500     C103        S  
15       0         248706  16.0000  Unknown        S  
..     ...            ...      ...      ...      ...  
866      0  SC/PARIS 2149  13.8583  Unknown        C  
874      0      P/PP 3381  24.0000  Unknown        C  
875      0           2667   7.2250  Unknown        C  
879      1          11767  83.1583      C50        C  
889      0         111369  30.0000     C148        C  

[217 rows x 12 columns]

#### 16) get the details of passengers in between 3 to 10 years old

In [31]:
df[(df['Age']>=3) & (df['Age']<=10)]

PassengerId  Survived  Pclass  \
10            11         1       3   
24            25         0       3   
43            44         1       2   
50            51         0       3   
58            59         1       2   
63            64         0       3   
147          148         0       3   
165          166         1       3   
171          172         0       3   
182          183         0       3   
184          185         1       3   
193          194         1       2   
233          234         1       3   
237          238         1       2   
261          262         1       3   
278          279         0       3   
348          349         1       3   
374          375         0       3   
407          408         1       2   
419          420         0       3   
445          446         1       1   
448          449         1       3   
480          481         0       3   
489          490         1       3   
535          536         1       2   
541          542         0       3   
549          550         1       2   
618          619         1       2   
634          635         0       3   
691          692         1       3   
720          721         1       2   
750          751         1       2   
751          752         1       3   
777          778         1       3   
787          788         0       3   
813          814         0       3   
819          820         0       3   
850          851         0       3   
852          853         0       3   
869          870         1       3   

                                                Name     Sex   Age  SibSp  \
10                   Sandstrom, Miss. Marguerite Rut  female   4.0      1   
24                     Palsson, Miss. Torborg Danira  female   8.0      3   
43          Laroche, Miss. Simonne Marie Anne Andree  female   3.0      1   
50                        Panula, Master. Juha Niilo    male   7.0      4   
58                      West, Miss. Constance Mirium  female   5.0      1   
63                             Skoog, Master. Harald    male   4.0      3   
147                 Ford, Miss. Robina Maggie "Ruby"  female   9.0      2   
165  Goldsmith, Master. Frank John William "Frankie"    male   9.0      0   
171                             Rice, Master. Arthur    male   4.0      4   
182            Asplund, Master. Clarence Gustaf Hugo    male   9.0      4   
184              Kink-Heilmann, Miss. Luise Gretchen  female   4.0      0   
193                       Navratil, Master. Michel M    male   3.0      1   
233                   Asplund, Miss. Lillian Gertrud  female   5.0      4   
237                 Collyer, Miss. Marjorie "Lottie"  female   8.0      0   
261                Asplund, Master. Edvin Rojj Felix    male   3.0      4   
278                               Rice, Master. Eric    male   7.0      4   
348           Coutts, Master. William Loch "William"    male   3.0      1   
374                       Palsson, Miss. Stina Viola  female   3.0      3   
407                   Richards, Master. William Rowe    male   3.0      1   
419                        Van Impe, Miss. Catharina  female  10.0      0   
445                        Dodge, Master. Washington    male   4.0      0   
448                   Baclini, Miss. Marie Catherine  female   5.0      2   
480                   Goodwin, Master. Harold Victor    male   9.0      5   
489            Coutts, Master. Eden Leslie "Neville"    male   9.0      1   
535                           Hart, Miss. Eva Miriam  female   7.0      0   
541             Andersson, Miss. Ingeborg Constanzia  female   9.0      4   
549                   Davies, Master. John Morgan Jr    male   8.0      1   
618                      Becker, Miss. Marion Louise  female   4.0      2   
634                               Skoog, Miss. Mabel  female   9.0      3   
691                               Karun, Miss. Manca  female   4.0      0   
720                Harper, Miss. Annie Jessie "Nina"  femal

#### 17) get the male passengers who were survived or female passengers who not survived.

In [32]:
df[((df['Sex']=='male') & (df['Survived']==1)) | ((df['Sex']=='female') & (df['Survived']==0))]

PassengerId  Survived  Pclass  \
14            15         0       3   
17            18         1       2   
18            19         0       3   
21            22         1       2   
23            24         1       1   
..           ...       ...     ...   
869          870         1       3   
882          883         0       3   
885          886         0       3   
888          889         0       3   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
14                Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   
17                        Williams, Mr. Charles Eugene    male  28.0      0   
18   Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
21                               Beesley, Mr. Lawrence    male  34.0      0   
23                        Sloper, Mr. William Thompson    male  28.0      0   
..                                                 ...     ...   ...    ...   
869                    Johnson, Master. Harold Theodor    male   4.0      1   
882                       Dahlberg, Miss. Gerda Ulrika  female  22.0      0   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female  28.0      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch      Ticket     Fare    Cabin Embarked  
14       0      350406   7.8542  Unknown        S  
17       0      244373  13.0000  Unknown        S  
18       0      345763  18.0000  Unknown        S  
21       0      248698  13.0000      D56        S  
23       0      113788  35.5000       A6        S  
..     ...         ...      ...      ...      ...  
869      1      347742  11.1333  Unknown        S  
882      0        7552  10.5167  Unknown        S  
885      5      382652  29.1250  Unknown        Q  
888      2  W./C. 6607  23.4500  Unknown        S  
889      0      111369  30.0000     C148        C  

[190 rows x 12 columns]

#### 18) get the passenger id,name and class who where embarked in queenstown and survived.

In [33]:
df[(df['Embarked']=='Q') & (df['Survived']==1)][['PassengerId','Name','Pclass']]

PassengerId                                             Name  Pclass
22            23                      McGowan, Miss. Anna "Annie"       3
28            29                    O'Dwyer, Miss. Ellen "Nellie"       3
32            33                         Glynn, Miss. Mary Agatha       3
44            45                    Devaney, Miss. Margaret Delia       3
47            48                        O'Driscoll, Miss. Bridget       3
82            83                   McDermott, Miss. Brigdet Delia       3
109          110                              Moran, Miss. Bertha       3
156          157                 Gilnagh, Miss. Katherine "Katie"       3
186          187  O'Brien, Mrs. Thomas (Johanna "Hannah" Godfrey)       3
198          199                 Madigan, Miss. Margaret "Maggie"       3
208          209                        Carr, Miss. Helen "Ellen"       3
241          242                   Murphy, Miss. Katherine "Kate"       3
274          275                       Healy, Miss. Hanora "Nora"       3
289          290                             Connolly, Miss. Kate       3
300          301         Kelly, Miss. Anna Katherine "Annie Kate"       3
301          302                               McCoy, Mr. Bernard       3
303          304                              Keane, Miss. Nora A       2
322          323                        Slayter, Miss. Hilda Mary       2
330          331                               McCoy, Miss. Agnes       3
358          359                             McGovern, Miss. Mary       3
359          360                Mockler, Miss. Helen Mary "Ellie"       3
368          369                              Jermyn, Miss. Annie       3
412          413                           Minahan, Miss. Daisy E       1
510          511                         Daly, Mr. Eugene Patrick       3
573          574                                Kelly, Miss. Mary       3
612          613                      Murphy, Miss. Margaret Jane       3
653          654                    O'Leary, Miss. Hanora "Norah"       3
697          698                 Mullens, Miss. Katherine "Katie"       3
727          728                         Mannion, Miss. Margareth       3
828          829                     McCormack, Mr. Thomas Joseph       3

#### 19) get the details of passengers who have siblings and spouses abroad is more than 2 or parents and children abroad is less than 2.

In [35]:
df[(df['SibSp']>2) | (df['Parch']<2)]

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
884          885         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
884                             Sutehall, Mr. Henry Jr    male  25.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare    Cabin Embarked  
0        0         A/5 21171   7.2500  Unknown        S  
1        0          PC 17599  71.2833      C85        C  
2        0  STON/O2. 3101282   7.9250  Unknown        S  
3        0            113803  53.1000     C123        S  
4        0            373450   8.0500  Unknown        S  
..     ...               ...      ...      ...      ...  
884      0   SOTON/OQ 392076   7.0500  Unknown        S  
886      0            211536  13.0000  Unknown        S  
887      0            112053  30.0000      B42        S  
889      0            111369  30.0000     C148        C  
890      0            370376   7.7500  Unknown        Q  

[824 rows x 12 columns]

#### 20) get the details of male passengers who were embarked in southampton.

In [36]:
df[(df['Sex']=='male') & (df['Embarked']=='S')]

PassengerId  Survived  Pclass                            Name   Sex  \
0              1         0       3         Braund, Mr. Owen Harris  male   
4              5         0       3        Allen, Mr. William Henry  male   
6              7         0       1         McCarthy, Mr. Timothy J  male   
7              8         0       3  Palsson, Master. Gosta Leonard  male   
12            13         0       3  Saundercock, Mr. William Henry  male   
..           ...       ...     ...                             ...   ...   
878          879         0       3              Laleff, Mr. Kristo  male   
881          882         0       3              Markun, Mr. Johann  male   
883          884         0       2   Banfield, Mr. Frederick James  male   
884          885         0       3          Sutehall, Mr. Henry Jr  male   
886          887         0       2           Montvila, Rev. Juozas  male   

      Age  SibSp  Parch            Ticket     Fare    Cabin Embarked  
0    22.0      1      0         A/5 21171   7.2500  Unknown        S  
4    35.0      0      0            373450   8.0500  Unknown        S  
6    54.0      0      0             17463  51.8625      E46        S  
7     2.0      3      1            349909  21.0750  Unknown        S  
12   20.0      0      0         A/5. 2151   8.0500  Unknown        S  
..    ...    ...    ...               ...      ...      ...      ...  
878  28.0      0      0            349217   7.8958  Unknown        S  
881  33.0      0      0            349257   7.8958  Unknown        S  
883  28.0      0      0  C.A./SOTON 34068  10.5000  Unknown        S  
884  25.0      0      0   SOTON/OQ 392076   7.0500  Unknown        S  
886  27.0      0      0            211536  13.0000  Unknown        S  

[441 rows x 12 columns]

#### 21) get the name, age, ticket, cabin no and fare of female passengers who were embarked in queenstown.

In [37]:
df[(df['Sex']=='female') & (df['Embarked']=='Q')][['Name','Age','Ticket','Cabin','Fare']]

Name   Age  Ticket    Cabin  \
22                       McGowan, Miss. Anna "Annie"  15.0  330923  Unknown   
28                     O'Dwyer, Miss. Ellen "Nellie"  28.0  330959  Unknown   
32                          Glynn, Miss. Mary Agatha  28.0  335677  Unknown   
44                     Devaney, Miss. Margaret Delia  19.0  330958  Unknown   
47                         O'Driscoll, Miss. Bridget  28.0   14311  Unknown   
82                    McDermott, Miss. Brigdet Delia  28.0  330932  Unknown   
109                              Moran, Miss. Bertha  28.0  371110  Unknown   
156                 Gilnagh, Miss. Katherine "Katie"  16.0   35851  Unknown   
186  O'Brien, Mrs. Thomas (Johanna "Hannah" Godfrey)  28.0  370365  Unknown   
198                 Madigan, Miss. Margaret "Maggie"  28.0  370370  Unknown   
208                        Carr, Miss. Helen "Ellen"  16.0  367231  Unknown   
241                   Murphy, Miss. Katherine "Kate"  28.0  367230  Unknown   
264                               Henry, Miss. Delia  28.0  382649  Unknown   
274                       Healy, Miss. Hanora "Nora"  28.0  370375  Unknown   
289                             Connolly, Miss. Kate  22.0  370373  Unknown   
300         Kelly, Miss. Anna Katherine "Annie Kate"  28.0    9234  Unknown   
303                              Keane, Miss. Nora A  28.0  226593     E101   
322                        Slayter, Miss. Hilda Mary  30.0  234818  Unknown   
330                               McCoy, Miss. Agnes  28.0  367226  Unknown   
358                             McGovern, Miss. Mary  28.0  330931  Unknown   
359                Mockler, Miss. Helen Mary "Ellie"  28.0  330980  Unknown   
368                              Jermyn, Miss. Annie  28.0   14313  Unknown   
412                           Minahan, Miss. Daisy E  33.0   19928      C78   
501                              Canavan, Miss. Mary  21.0  364846  Unknown   
502                   O'Sullivan, Miss. Bridget Mary  28.0  330909  Unknown   
573                                Kelly, Miss. Mary  28.0   14312  Unknown   
593                               Bourke, Miss. Mary  28.0  364848  Unknown   
612                      Murphy, Miss. Margaret Jane  28.0  367230  Unknown   
653                    O'Leary, Miss. Hanora "Norah"  28.0  330919  Unknown   
654                     Hegarty, Miss. Hanora "Nora"  18.0  365226  Unknown   
657                    Bourke, Mrs. John (Catherine)  32.0  364849  Unknown   
680                              Peters, Miss. Katie  28.0  330935  Unknown   
697                 Mullens, Miss. Katherine "Katie"  28.0   35852  Unknown   
727                         Mannion, Miss. Margareth  28.0   36866  Unknown   
767                               Mangan, Miss. Mary  30.5  364850  Unknown   
885             Rice, Mrs. William (Margaret Norton)  39.0  382652  Unknown   

        Fare  
22    8.0292  
28    7.8792  
32    7.7500  
44    7.8792  
47    7.7500  
82    7.7875  
109  24.1500  
156   7.7333  
186  15.5000  
198   7.7500  
208   7.7500  
241  15.5000  
264   7.7500  
274   7.7500  
289   7.7500  
300   7.7500  
303  12.3500  
322  12.3500  
330  23.2500  
358   7.8792  
359   7.8792  
368   7.7500  
412  90.0000  
501   7.7500  
502   7.6292  
573   7.7500  
593   7.7500  
612  15.5000  
653   7.8292  
654   6.7500  
657  15.5000  
680   8.1375  
697   7.7333  
727   7.7375  
767   7.7500  
885  29.1250

#### 22) get the details of passengers in first class.

In [38]:
df[df['Pclass']==1]

PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
6              7         0       1   
11            12         1       1   
23            24         1       1   
..           ...       ...     ...   
871          872         1       1   
872          873         0       1   
879          880         1       1   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
23                        Sloper, Mr. William Thompson    male  28.0      0   
..                                                 ...     ...   ...    ...   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
872                           Carlsson, Mr. Frans Olof    male  33.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch    Ticket     Fare        Cabin Embarked  
1        0  PC 17599  71.2833          C85        C  
3        0    113803  53.1000         C123        S  
6        0     17463  51.8625          E46        S  
11       0    113783  26.5500         C103        S  
23       0    113788  35.5000           A6        S  
..     ...       ...      ...          ...      ...  
871      1     11751  52.5542          D35        S  
872      0       695   5.0000  B51 B53 B55        S  
879      1     11767  83.1583          C50        C  
887      0    112053  30.0000          B42        S  
889      0    111369  30.0000         C148        C  

[216 rows x 12 columns]

#### 23) get the details of female passengers who have more than 3 siblings and spouses abroad.

In [39]:
df[(df['Sex']=='female') & (df['SibSp']>3)]

PassengerId  Survived  Pclass                                  Name  \
68            69         1       3       Andersson, Miss. Erna Alexandra   
71            72         0       3            Goodwin, Miss. Lillian Amy   
119          120         0       3     Andersson, Miss. Ellis Anna Maria   
180          181         0       3          Sage, Miss. Constance Gladys   
233          234         1       3        Asplund, Miss. Lillian Gertrud   
541          542         0       3  Andersson, Miss. Ingeborg Constanzia   
542          543         0       3     Andersson, Miss. Sigrid Elisabeth   
792          793         0       3               Sage, Miss. Stella Anna   
813          814         0       3    Andersson, Miss. Ebba Iris Alfrida   
863          864         0       3     Sage, Miss. Dorothy Edith "Dolly"   

        Sex   Age  SibSp  Parch    Ticket     Fare    Cabin Embarked  
68   female  17.0      4      2   3101281   7.9250  Unknown        S  
71   female  16.0      5      2   CA 2144  46.9000  Unknown        S  
119  female   2.0      4      2    347082  31.2750  Unknown        S  
180  female  28.0      8      2  CA. 2343  69.5500  Unknown        S  
233  female   5.0      4      2    347077  31.3875  Unknown        S  
541  female   9.0      4      2    347082  31.2750  Unknown        S  
542  female  11.0      4      2    347082  31.2750  Unknown        S  
792  female  28.0      8      2  CA. 2343  69.5500  Unknown        S  
813  female   6.0      4      2    347082  31.2750  Unknown        S  
863  female  28.0      8      2  CA. 2343  69.5500  Unknown        S

#### 24) get the male passenger details who either younger than 18 and older than 60.

In [40]:
df[(df['Sex']=='male') & ((df['Age']<18) | (df['Age']>60))]

PassengerId  Survived  Pclass                                     Name  \
7              8         0       3           Palsson, Master. Gosta Leonard   
16            17         0       3                     Rice, Master. Eugene   
33            34         0       2                    Wheadon, Mr. Edward H   
50            51         0       3               Panula, Master. Juha Niilo   
54            55         0       1           Ostby, Mr. Engelhart Cornelius   
..           ...       ...     ...                                      ...   
841          842         0       2                 Mudd, Mr. Thomas Charles   
844          845         0       3                      Culumovic, Mr. Jeso   
850          851         0       3  Andersson, Master. Sigvard Harald Elias   
851          852         0       3                      Svensson, Mr. Johan   
869          870         1       3          Johnson, Master. Harold Theodor   

      Sex   Age  SibSp  Parch       Ticket     Fare    Cabin Embarked  
7    male   2.0      3      1       349909  21.0750  Unknown        S  
16   male   2.0      4      1       382652  29.1250  Unknown        Q  
33   male  66.0      0      0   C.A. 24579  10.5000  Unknown        S  
50   male   7.0      4      1      3101295  39.6875  Unknown        S  
54   male  65.0      0      1       113509  61.9792      B30        C  
..    ...   ...    ...    ...          ...      ...      ...      ...  
841  male  16.0      0      0  S.O./P.P. 3  10.5000  Unknown        S  
844  male  17.0      0      0       315090   8.6625  Unknown        S  
850  male   4.0      4      2       347082  31.2750  Unknown        S  
851  male  74.0      0      0       347060   7.7750  Unknown        S  
869  male   4.0      1      1       347742  11.1333  Unknown        S  

[77 rows x 12 columns]

#### 25) get the details of passengers who were in second class, older than 50 and parents/children abroad.

In [41]:
df[(df['Pclass']==2) & (df['Age']>50) & (df['Parch']>0)]

PassengerId  Survived  Pclass                                   Name  \
684          685         0       2      Brown, Mr. Thomas William Solomon   
774          775         1       2  Hocking, Mrs. Elizabeth (Eliza Needs)   

        Sex   Age  SibSp  Parch Ticket  Fare    Cabin Embarked  
684    male  60.0      1      1  29750  39.0  Unknown        S  
774  female  54.0      1      3  29105  23.0  Unknown        S